tooling/support/core/src/main/kotlin/com/squareup/tooling/support/core/extractors/DependencyExtractors.kt

```diff
@@ -20,21 +20,74 @@ package com.squareup.tooling.support.core.extractors
 import com.squareup.tooling.models.SquareDependency
 import com.squareup.tooling.support.core.models.SquareDependency
 import org.gradle.api.Project
+import org.gradle.api.artifacts.Configuration
 import org.gradle.api.artifacts.ConfigurationContainer
 import org.gradle.api.artifacts.Dependency
 import org.gradle.api.artifacts.ProjectDependency
+import org.gradle.api.artifacts.result.ResolvedDependencyResult
+import org.gradle.api.internal.artifacts.DefaultProjectComponentIdentifier
 import org.gradle.api.internal.artifacts.dependencies.AbstractExternalModuleDependency
 import org.gradle.api.internal.artifacts.dependencies.AbstractModuleDependency
 
-// Extracts Gradle Dependency objects from the given configurations
-public fun ConfigurationContainer.extractDependencies(
+public fun ConfigurationContainer.extractSquareDependencies(
+  project: Project,
   vararg configurationNames: String
-): Sequence<Dependency> {
-  return configurationNames.asSequence().flatMap { configurationName ->
-    getByName(configurationName).allDependencies.asSequence()
-  }
+): Sequence<SquareDependency> {
+  return configurationNames.asSequence()
+    .map { configurationName -> getByName(configurationName) }
+    .flatMap { configuration ->
+      sequence {
+        yieldAll(configuration.extractDependencies()
+          .map { it.extractSquareDependency(project) })
+        yieldAll(configuration.extractResolvedProjectDependencies())
```

Comment:
I wonder if we can put this behind a flag for `includedBuild` uses? Although, I believe there can be other cases of dependency substitution being used that doesn't involve `includedBuild`.

Reply:
Sounds good, I added it as a custom gradle property based on `useIncludeBuild`.